src/nginx/grpc_server_call.cc

```diff
@@ -358,11 +361,9 @@ void NgxEspGrpcServerCall::OnHttpBlockReading(ngx_http_request_t *r) {
         "NgxEspGrpcServerCall::OnHttpBlockReading: connection error");
 
     ngx_esp_request_ctx_t *ctx = ngx_http_esp_ensure_module_ctx(r);
-    NgxEspGrpcServerCall *server_call = ctx->grpc_server_call;
-    if (server_call != nullptr && server_call->grpc_upstream_cancel_) {
-      std::function<void()> grpc_upstream_cancel;
-      std::swap(grpc_upstream_cancel, server_call->grpc_upstream_cancel_);
-      grpc_upstream_cancel();
+    if (ctx->grpc_upstream_cancel) {
+      // std::swap(grpc_upstream_cancel, ctx->grpc_upstream_cancel);
+      (*ctx->grpc_upstream_cancel)();
```

Comment:
You have to free it.

ctx->grpc_upstream_cacnel.reset()

Reply:
Done -- I instead released it first then used it.